Templates/scu.json

```diff
@@ -0,0 +1,4764 @@
+      "fields": [
+        { "name": "TargetList", "type": "array", "subtype": "object_reference" },
+        { "name": "Trigger", "type": "ObjEventSetterSpawner::Trigger" },
+        { "name": "Condition", "type": "ObjEventSetterSpawner::Condition" },
+        { "name": "TimesType", "type": "ObjEventSetterSpawner::TimesType" },
+        { "name": "EventList", "type": "array", "subtype": "object_reference" },
+        { "name": "EventNotifier", "type": "ObjEventSetterSpawner::EventNotifier" },
+        { "name": "EventTimer", "type": "float32" },
+        { "name": "EventList2", "type": "array", "subtype": "object_reference" },
+        { "name": "EventNotifier2", "type": "ObjEventSetterSpawner::EventNotifier2" },
+        { "name": "EventTimer2", "type": "float32" },
+        { "name": "EventList3", "type": "array", "subtype": "object_reference" },
+        { "name": "EventNotifier3", "type": "ObjEventSetterSpawner::EventNotifier3" },
+        { "name": "EventTimer3", "type": "float32" }
+      ]
+    },
+    "ObjHintRingSpawner": {
+      "fields": [
+        { "name": "HintID", "type": "uint32" },
+        { "name": "IsMessageBoot", "type": "bool" },
+        { "name": "OnlyOneBoot", "type": "bool" }
+      ]
+    },
+    "ObjHintRingMultiSpawner": {
+      "fields": [
+        { "name": "HintID", "type": "uint32" },
+        { "name": "HintID2", "type": "uint32" },
+        { "name": "HintID3", "type": "uint32" },
+        { "name": "IsMessageBoot", "type": "bool" },
+        { "name": "OnlyOneBoot", "type": "bool" }
+      ]
+    },
+    "ObjGrouperSpawner": {
+      "fields": [
+        { "name": "TargetList", "type": "array", "subtype": "object_reference" },
+        { "name": "Connect", "type": "bool" }
+      ]
+    },
+    "ObjLayerBoardSpawner": {
+      "fields": [
+        { "name": "CollisionWidth", "type": "float32" },
+        { "name": "CollisionHeight", "type": "float32" },
+        { "name": "OnLayer0", "type": "uint8" },
+        { "name": "OnLayer1", "type": "uint8" },
+        { "name": "OffLayer0", "type": "uint8" },
+        { "name": "OffLayer1", "type": "uint8" },
+        { "name": "IsBackEffective", "type": "bool" }
+      ]
+    },
+    "ObjPointMarkerSpawner": {
+      "fields": [
+        { "name": "Width", "type": "float32" },
+        { "name": "Height", "type": "float32" },
+        { "name": "StageType", "type": "ObjPointMarkerSpawner::StageType" }
+      ]
+    },
+    "ObjShadowSpawner": {
+      "fields": [
+        { "name": "Width", "type": "float32" },
+        { "name": "Height", "type": "float32" },
+        { "name": "Depth", "type": "float32" }
+      ]
+    },
+    "ObjTS_CheckpointSpawner": {
+      "fields": [
+        { "name": "CollisionWidth", "type": "float32" },
+        { "name": "CollisionHeight", "type": "float32" },
+        { "name": "CollisionLength", "type": "float32" },
+        { "name": "NextID", "type": "object_reference", "reftype": "float32" }
+      ]
+    },
```

Comment:
Exclusive object to Sonic Colors: Ultimate. A `reftype` parameter needs to be added due to `NextID` using a `float32` for the object id of the next object for some reason...

Reply:
The parameter type for `NextID` has been changed to a simple `float32` as the set data parser function doesn't read that object parameter nor does it ever hit a data breakpoint. With this in mind there is no reason to spend extra development efforts on the addition of a `reftype` attribute or some other solution and can be just kept as a float32 as it literally does not matter what value this field has set.